### PR TITLE
Improve e2e test reliability

### PR DESCRIPTION
**Pull request by JorTurFer**
Signed-off-by: jorturfer <jorge_turrado@hotmail.es>

This PR increases some timeouts across different e2e test and does these relevant changes:
- **Azure Pipelines**: e2e tests now check scaling from/to 0 instead of between 1 and 3 because otherwise we have random fails if the removed pods during the scale in are still executing an AzDO job.
- **New Relic**: e2e test has changed a bit to maintain the load till the scaling out is done.
- **Selenium**: e2e test has changed, now nodes starts from 0 and check scaling to 0 instead of job status because sometimes the job started before the hub is ready and has more than 1 result (and the test fails)
- **Global**: The timeout for each scaler e2e tests has been increased from 10m to 30m.
- **Global**: Removed all the resources except for cpu e2e tests
- **Global**: Failing e2e tests are retried one more time

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated

### CODE REVIEW

**Comment by tomkerkhove** [comment]
Sorry but I cancelled the test run so that #2583 can be triggered, verified and merged for our release first.

/run-e2e

**Comment by JorTurFer** [comment]
no problem

**Review by tomkerkhove — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,7 +36,7 @@
 
 ### Other
 
-- TODO ([#XXX](https://github.com/kedacore/keda/pull/XXX))
+- Improve e2e tests reliability ([#2580](https://github.com/kedacore/keda/issues/2580))
```
[the remark this answers is not in the document]
Comment (JorTurFer) [reply CHANGELOG.md:39]: sure, nice catch

**Review by zroubalik — APPROVED** [review]
Incredible work 🚀

**Comment by zroubalik** [comment]
@JorTurFer would you mind resolving those conflicts, we can merge it then!

**Comment by tomkerkhove** [comment]
Thanks a ton @JorTurFer!

**Comment by JorTurFer** [comment]
you're welcome ❤️

**Pull request merged**